src/main.rs

```diff
@@ -72,11 +77,114 @@ impl fmt::Display for Sample {
     }
 }
 
+/// Define a struct to carry the information about the process
+/// to track. The process can be either external or internal.
+///
+/// This enum dereferences to the psutil::Process to gather information 
+/// about system usage.
+pub enum TrackedProcess {
+  /// An external process was started outside of this program and
+  /// submitted using the --pid parameter.
+  External(Process),
+  /// An internal process is started by procrec as a fork and requires
+  /// joining the forked process.
+  Internal(Process, std::process::Child)
+}
+
+impl<'a> TryFrom<&'a Opts> for TrackedProcess {
+    type Error = String;
+
+    fn try_from(opts: &'a Opts) -> Result<Self, Self::Error> {
+     match opts.pid {
+       Some(pid) => match Process::new(pid) {
+         Ok(p) => Ok(TrackedProcess::External(p)),
+         Err(e) => Err(format!("Failed accessing process: {}", e))
+       },
+       None => {
+         let cl = &opts.command;
+         if cl.len() == 0 {
+           return Err("Process to record must be provided as additional argument or via '--pid' parameter. For detailed information, execute with --help".to_owned())
+         }
+           
+         // Create the command line for the process to be executed
+         let mut cmd = Command::new(cl[0].clone());
+         if cl.len() > 1 {
+           cmd.args(&cl[1..]);
+         }
+       
+         match cmd.spawn() {
+           Ok(c) => match Process::new(c.id()) {
```

Comment:
Same as above:
I'm still not sure if these nested matches are the most idiomatic and maintainable way.
another option would be something like `opts.unwrap_or_else(|e| panic!("err: {}", e))` or when you are directly returning a result again just `map` it from one `Ok` to another.
If the error handling gets cumbersome adding `anyhow` crate could be an option, even though I'd like to keep the required deps to a minimum.